src/Table/Table.tsx

```diff
@@ -0,0 +1,984 @@
+                                                <div
+                                                    className={"icon-holder" + (row.expanded ? " active" : "")}
+                                                    onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => { props.onRowExpanded(e, row); }}
+                                                >
+                                                    {row.expanded ? angleDown : angleRightIcon}
+                                                </div>
+                                            }
+                                        </td>
+                                        :
+                                        ((row.subRows.length > 0 || row.rowContentDetail) && props.rowsAreCollapsable) &&
+                                        <td>
+                                            <div
+                                                className={"icon-holder" + (row.expanded ? " active" : "")}
+                                                onClick={(e: React.MouseEvent<HTMLDivElement, MouseEvent>) => { props.onRowExpanded(e, row); }}
+                                            >
+                                                {row.expanded ? angleDown : angleRightIcon}
+                                            </div>
+                                        </td>
+                                    }
+                                    {row.cells.map((cell: Cell, cellIndex: number) => {
+                                        return <td key={`cell-${cellIndex}`}>
+                                            {cell.value}
+                                        </td>;
+                                    })}
+                                    {props.useShowActionColumn &&
+                                        <td>
+                                            <ActionColumn
+                                                actionLinks={props.actionLinks}
+                                                primaryActionButton={props.primaryActionButton}
+                                                selectedRow={row}
+                                                tableRef={tableRef}
+                                                onActionDropped={(event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+                                                    props.onActionDropped(event, row, null);
+                                                }}
+                                            />
+                                        </td>
+                                    }
+                                </tr>
+
+                                {row.subRows.map((subRow: TableRow) => {
+                                    const checkSubRowRandomIds: string = randomId("sub-chk-");
+                                    return (
+                                        <React.Fragment key={`sub-row-${subRow.rowIndex}`}>
```

Comment:
The row and subrow parts looks like duplicated code, it will be better to create a tableRow component that can act as a row or a subrow.

Reply:
Thought about that, you might noticed i took the actionComponent out and make it into another FC. However, while they look familiar. They are kind of different in so many ways ranging from classNames to styling and paddings . While i agree with you that they have a lot in common.  Reusing a component for the two will introduce complications and will make it difficult to maintain. Maintenance-wise, the verbose version is clearer.

Reply:
We will do this refactoring in another PR @omarVodiaK .